### PR TITLE
feat(gpio_lib): add basic gpio library implementation

### PR DESCRIPTION
**Pull request by Moleus**

### CODE REVIEW

**Comment by Moleus** [comment]
Please, don't merge this one until it builds successfully

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,15 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.11)
 
 project(motorland C)
 
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (Moleus) [inline CMakeLists.txt:6]: It's for c++, replace with C

**Review by Moleus — COMMENTED** [review]
Fix minor cmake mistakes

File: proto_libs/gpio_lib/include/gpio_lib.h
Comment (filberol) [inline proto_libs/gpio_lib/include/gpio_lib.h:1]: delete?

**Review by filberol — APPROVED** [review]
Everything ok, integration test reviewed

**Pull request merged**